src/lib.rs

```diff
@@ -126,8 +139,90 @@ pub fn select_coin_knapsack(
 fn knap_sack(
     adjusted_target: u64,
     smaller_coins: &[(usize, OutputGroup)],
+    inputs: &[OutputGroup],
+    options: CoinSelectionOpt,
 ) -> Result<SelectionOutput, SelectionError> {
-    unimplemented!()
+    let mut target_reached = false;
+    let mut selected_inputs: HashSet<(usize)> = HashSet::new();
+    let mut accumulated_value: u64 = 0;
+    let mut accumulated_weight: u32 = 0;
```

Comment:
This variable need not be tracked.
The weight of the selected inputs can be calculated once the best_set is computed, i.e when you return the result

Reply:
If we calculate best_set_weight after the loops, we will have to run another loop, iterating through smaller_coins, `for &(index, u) in smaller_coins {`  matching best_set with it and extracting weights from them. Would that be preferable ?

Reply:
When you compute the weight at the end you are only computing the weight of the best_set exactly once.
In the current implementation you are computing weight even for the selections that might not result in a best_set.
So the current implementation is less efficient.